email/email.c

```diff
@@ -159,3 +159,81 @@ int emaillist_add_email(struct EmailList *el, struct Email *e)
 
   return 0;
 }
+
+/**
+ * header_find - Find a header, matching on its field, in a list of headers
+ * @param hdrlist List of headers to search
+ * @param header  The header to search for
+ * @retval node   The node in the list matching the header
+ * @retval NULL   If no matching header is found
+ *
+ * The header should either of the form "X-Header:" or "X-Header: value"
+ */
+struct ListNode *header_find(const struct ListHead *hdrlist, const char *header)
+{
+  struct ListNode *n = NULL;
+
+  const int keylen = strchr(header, ':') - header + 1;
```

Comment:
Please can you split this up.
Check that `strchr()` succeeds before doing pointer maths.

Reply:
done 1676ed1